process/dataindexer/dataIndexer.go

```diff
@@ -77,21 +75,55 @@ func (di *dataIndexer) SaveBlock(outportBlock *outport.OutportBlock) error {
 		return err
 	}
 
-	wi := workItems.NewItemBlock(
-		di.elasticProcessor,
-		&outport.OutportBlockWithHeader{
-			OutportBlock: outportBlock,
-			Header:       header,
-		},
-	)
-	di.dispatcher.Add(wi)
+	headerHash := outportBlock.BlockData.HeaderHash
```

Comment:
all the new code could have been extracted into a function

Reply:
done